ci/webhook-receiver.py

```diff
@@ -51,54 +135,83 @@ def filter_to_prs_and_pr_comments(json):
 def init_ci_pipeline():
-
-    extracted_data = filter_to_prs_and_pr_comments(request.json)
-    if extracted_data is not None:
-        print("Data matches filter, we should RUN CI")
-        print(extracted_data)
-
-        # Check if there is another CI run in progress
-        proc1 = subprocess.Popen(['ps', 'cax'], stdout=subprocess.PIPE)
-        proc2 = subprocess.Popen(['grep', 'manager.sh'], stdin=proc1.stdout,
-                                         stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        proc1.stdout.close()
-        out, err = proc2.communicate()
-
-        if (out):
-            print("Can't run CI, another CI run in progress")
-            os.system("./ci_busy.sh config {} \"{}\" \"Another CI run in progress, please try again in 15 minutes\""
-                      .format(extracted_data['id'], extracted_data['body']))
-        else:
-            os.system("./manager.sh config {} \"{}\"".format(extracted_data['id'], extracted_data['body']))
+    request_ctx = filter_to_prs_and_pr_comments(request.json)
+    if request_ctx is None:
+        logging.debug("Request filter doesn't match request")
+        return jsonify({"success": True})
+    
+    request_ctx['src_ip'] = ip_address(u'{}'.format(request.access_route[0]))
+    request_ctx['X-Hub-Signature'] = request.headers.get('X-Hub-Signature')
+    request_ctx['data'] = request.data
+
+    if not verify_request_ip(request_ctx):
+        print("Incoming webhook not from a valid Github address")
+        log_access_denied(request_ctx, "Incoming webhook not from a valid Github address")
+        return jsonify({"success": True})
+
+    if not verify_request_secret(request_ctx):
+        print("Incoming webhook secret doesn't match configured secret")
+        log_access_denied(request_ctx, "Incoming webhook has an invalid secret")
+        return jsonify({"success": True})
+
+    if (request_ctx['repo'] == 'openNetVM' and request_ctx['user'] not in authorized_users):
+        print("Incoming request is from an unathorized user")
+        log_access_denied("Incoming request is from an unathorized user")
+        os.system("./ci_busy.sh config {} \"{}\" \"{}\" \"User not authorized to run CI, please contact one of the repo maintainers\""
+                  .format(request_ctx['id'], request_ctx['repo'], request_ctx['body']))
+        return jsonify({"success": True})
+
+    print("Request matches filter, we should RUN CI. {}".format(get_request_info(request_ctx)))
+
+    # Check if there is another CI run in progress
+    proc1 = subprocess.Popen(['ps', 'cax'], stdout=subprocess.PIPE)
+    proc2 = subprocess.Popen(['grep', 'manager.sh'], stdin=proc1.stdout,
+                                     stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    proc1.stdout.close()
+    out, err = proc2.communicate()
+
+    if (out):
+        print("Can't run CI, another CI run in progress")
+        log_access_granted(request_ctx, "CI busy, posting busy msg")
```

Comment:
Access granted will be printed more often than denied.  What happens if there is no more disk space for the server to write logs to?  Will the server block?  Will it die?  Can we make access granted logs be printed only if we have a switch in the config file?

Regardless, still worth thinking about log rotation strategy -- how can we leverage either (a) [logrotate](https://linux.die.net/man/8/logrotate) or (b) have the server do something to manage log size?